HOTT-1195 coloured breadcrumb

jebw opened the pull request:
### Jira link

[HOTT-1195](https://transformuk.atlassian.net/browse/HOTT-1195)

### What?

I have added/removed/altered:

- [x] Added a 'colour' option to the segmented comm codes
- [x] Added a helper to shorten comm codes
- [x] Added the Azeret Mono font into webpacker, together with a common `_fonts.scss` definition file, much like the existing `_colours.scss`
- [x] Added a tree like view of the declarable's ancestry, using `aria-owns` attribute to define the relationships
- [x] Removed the background shaded panel on country picker
- [x] Removed the ancestors from the classification row above the ancestor list now that we have the ancestor list

### Why?

I am doing this because:

- our users value the hierarchical view of the commodity ancestry and want it to return in a more accessible form

### Have you? (optional)

- [ ] Reviewed view changes with stake holders
- [x] Validated mobile responsive behaviour of view changes

### Deployment risks (optional)

- low technical risk but high user visibility

### Screenshots

![Screenshot from 2021-12-14 14-25-28](https://user-images.githubusercontent.com/10818/146016982-8fbbf4b4-d00e-40c1-bf02-879dc77ed473.png)
![Screenshot from 2021-12-14 14-24-34](https://user-images.githubusercontent.com/10818/146016990-b3f56a5d-2411-4c15-8a4a-f4b76e055cbc.png)
![Screenshot from 2021-12-14 14-24-15](https://user-images.githubusercontent.com/10818/146016991-ba202bbb-1ce4-4020-ba46-58add94a3244.png)

jebw committed:
HOTT-1195 Extended comm code helper to show colours

jebw committed:
HOTT-1195 Added helper to abbreviate comm codes

jebw committed:
HOTT-1195 Added commodity ancestor tree

jebw committed:
HOTT-1195 Added Azeret Mono font

jebw committed:
HOTT-1195 Added mobile overrides

jebw committed:
HOTT-1195 Use govuk frontend for sizes and colours

jebw committed:
HOTT-1195 Added aria labelling

jebw committed:
HOTT-1195 Show description of commodity in table

And removed the now redundent `classification_description` helper

jebw committed:
HOTT-1195 Removed background on country picker

jebw committed:
HOTT-1195 Added view spec for ancestors tree

jebw committed:
HOTT-1195 Show commodity tree for declarable headings

jebw committed:
HOTT-1195 Mobile fixes

1. Reduce indents by half
2. Fix font-size for Section/Chapter/Heading text

jebw committed:
HOTT-1195 Use chapter description for chapter row

alexdesi commented on app/helpers/commodities_helper.rb:
```diff
@@ -110,13 +110,28 @@ def declarable_heading_full(commodity)
+    else
+      code
+    end
+  end
```
not sure if it can help,
I wrote the `commodity_code_for_check_duties_service` : 
https://github.com/trade-tariff/trade-tariff-frontend/blob/b660f45024ed63fd7692d1550fa44344c03e9518/app/models/concerns/declarable.rb#L77

but maybe the logic is slightly different.

jebw replied:
I think I'll standardise this in a separate PR

alexdesi commented on app/views/commodities/_ancestors.html.erb:
```diff
@@ -0,0 +1,102 @@
+
+      <%# list each of commodities ancestors %>
+      <% declarable.ancestors.each.with_index do |ancestor, index| %>
+      <li id="commodity-ancestors__ancestor-<%= index + 1 %>"
```
it would be nice to have `commodity-ancestors__ancestor-?`  in a helper,
that is used in many places, 
also, should we check a max index?

jebw replied:
done

alexdesi commented on spec/views/commodities/_ancestors.html.erb_spec.rb:
```diff
@@ -0,0 +1,54 @@
+
+  context 'with commodity' do
+    let :declarable do
+      VCR.use_cassette 'commodities_2208909110' do
```
often we specify the cassette in the test or context 
`context 'with commodity', vcr: { cassette_name: ' ... '} do`

it seems a bit cleaner, but I not very opinionated on that anyway

jebw replied:
done

alexdesi reviewed (approved):
the PR with lot's of html/css are difficult to review, 
this looks good to me anyway

jebw committed:
HOTT-1195 Addressed review comments

1. Moved VCR setup to context methods
2. Use a helper to generate the ancestor element ids

jebw committed:
HOTT-1195 Last css tweaks
